File: java-api/sample/src/main/java/ovh/vicart/messaging/sample/viewmodel/RootViewModel.java
package ovh.vicart.messaging.sample.viewmodel;

import javafx.beans.property.SimpleListProperty;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import ovh.vicart.messaging.MessagingAPI;
import ovh.vicart.messaging.models.User;

import java.util.stream.Collectors;

public class RootViewModel {

    private final ObservableList<String> observableOnlineUsers = FXCollections.observableArrayList();
    private final MessagingAPI api;

    public RootViewModel() {
        api = MessagingAPI.getInstanceFor("tessske");
        api.getOnlineUsers().thenAccept(list -> {
            observableOnlineUsers.addAll(list.stream().map(User::getUsername).collect(Collectors.toList()));
        });
    }

    public SimpleListProperty<String> getOnlineUsers() {
        return new SimpleListProperty<>(observableOnlineUsers);
    }
}

File: java-api/sample-kt/src/main/kotlin/ovh/vicart/messagingkt/sample/viewmodel/RootViewModel.kt
package ovh.vicart.messagingkt.sample.viewmodel

import javafx.beans.binding.Bindings
import javafx.beans.property.SimpleIntegerProperty
import javafx.beans.property.SimpleListProperty
import javafx.beans.property.SimpleStringProperty
import javafx.collections.FXCollections
import kotlinx.coroutines.*
import ovh.vicart.messaging.models.User
import ovh.vicart.messagingkt.MessagingAPI

class RootViewModel {

    private lateinit var api: MessagingAPI

    private lateinit var users: List<User>
    private val observableOnlineUsers = FXCollections.observableArrayList<String>()
    val onlineUsers: SimpleListProperty<String> = SimpleListProperty(observableOnlineUsers)

    val selectedUserIndex = SimpleIntegerProperty(-1)
    val currentUsername = SimpleStringProperty()

    val messages = SimpleStringProperty("")

    init {
        selectedUserIndex.addListener { _, _, newValue ->
            messages.set("")
            CoroutineScope(Dispatchers.Default).launch {
                val msgs = api.getMessages(users[newValue.toInt()])
                println(msgs)
                messages.set(msgs.map { it.message }.joinToString("\n"))
            }
        }
    }

    fun loginInAPI(username: String) {
        api = MessagingAPI.getInstanceFor(username)
    }

    fun fetchOnlineUsers() {
        CoroutineScope(Dispatchers.Default).launch {
            users = api.getOnlineUsers()
            observableOnlineUsers.setAll(users.map { it.username })
        }
    }
}